pytgcalls/pytgcalls/mtproto/telethon_bridge.py

```diff
@@ -137,21 +137,27 @@ async def edit_group_call_member(self, peer, volume: int = None, muted=False):
         self.client._handle_update(response)
 
     async def get_and_set_self_peer(self):
-        self.my_peer = await self.client.get_me()
+        self.my_peer = await self.client.get_me(input_peer=True)
 
         return self.my_peer
 
     async def get_and_set_group_call(self, group):
+
         self.chat_peer = group
-        # TODO add if cond
-        self.chat_peer = await self.client.get_input_entity(group)
-        # TODO add more if cond
-        self.full_chat = (await self.client(functions.channels.GetFullChannelRequest(group))).full_chat
+
+        if type(group) not in [InputPeerChannel, InputPeerChat]:
+            self.chat_peer = await self.client.get_input_entity(group)
+
+        if isinstance(self.chat_peer, InputPeerChannel):
+            self.full_chat = (await self.client(functions.channels.GetFullChannelRequest(group))).full_chat
+        elif isinstance(self.chat_peer, InputPeerChat):
+            self.full_chat = (await self.client(functions.messages.GetFullChatRequest(group))).full_chat
 
         if self.full_chat is None:
             raise RuntimeError(f'Can\'t get full chat by {group}')
 
         self.group_call = self.full_chat.call
+        self.join_as = self.full_chat.groupcall_default_join_as
```

Comment:
Why u leave this row here? We already got it in resolve_and_set_join_as method

Reply:
Oupsy. Thanks for pointing out. It was left from the older commit

Reply:
Tho wouldn't be better if the default join was set by default?